library/src/test/java/com/yayandroid/locationmanager/providers/locationprovider/GooglePlayServicesLocationProviderTest.java

```diff
@@ -62,74 +81,72 @@ public void setUp() throws Exception {
     }
 
     @Test
-    public void onResumeShouldNotConnectGoogleApiClientWhenSettingsDialogIsOnTrue() {
+    public void onResumeShouldNotRequestLocationUpdateWhenSettingsDialogIsOnTrue() {
         makeSettingsDialogIsOnTrue();
 
         googlePlayServicesLocationProvider.onResume();
 
-        verify(mockedSource, never()).connectGoogleApiClient();
+        verify(mockedSource, never()).requestLocationUpdate();
     }
 
     @Test
-    public void onResumeShouldNotConnectWhenLocationIsAlreadyProvidedAndNotRequiredToKeepTracking() {
+    public void onResumeShouldNotRequestLocationUpdateWhenLocationIsAlreadyProvidedAndNotRequiredToKeepTracking() {
         when(locationConfiguration.keepTracking()).thenReturn(false);
 
         googlePlayServicesLocationProvider.onResume();
 
-        verify(mockedSource, never()).clearGoogleApiClient();
+        verify(mockedSource, never()).requestLocationUpdate();
     }
 
     @Test
-    public void onResumeShouldConnectWhenLocationIsNotYetProvided() {
+    public void onResumeShouldRequestLocationUpdateWhenLocationIsNotYetProvided() {
+        when(mockedSource.getLastLocation()).thenReturn(new MockSimpleTask<>(((Location) null)));
```

Comment:
This is not clear what you expect Task to do. See my comment on `MockSimpleTask`

Reply:
Done